File: gcc/testsuite/gfortran.dg/gomp/groupprivate-1.f90
module m
 implicit none
 integer :: ii
 integer :: x, y(20), z, v, u, k
 ! { dg-message "sorry, unimplemented: 'k' with 'omp groupprivate' on the host; try 'device_type\\(nohost\\)'" "" { target *-*-* } .-1 }
 ! { dg-message "sorry, unimplemented: 'u' with 'omp groupprivate' on the host; try 'device_type\\(nohost\\)'" "" { target *-*-* } .-2 }
 ! { dg-message "sorry, unimplemented: 'x' with 'omp groupprivate' on the host; try 'device_type\\(nohost\\)'" "" { target *-*-* } .-3 }
 ! { dg-message "sorry, unimplemented: 'y' with 'omp groupprivate' on the host; try 'device_type\\(nohost\\)'" "" { target *-*-* } .-4 }
 ! { dg-message "sorry, unimplemented: 'z' with 'omp groupprivate' on the host; try 'device_type\\(nohost\\)'" "" { target *-*-* } .-5 }

 ! { dg-prune-output "sorry, unimplemented: 'v' with 'omp groupprivate' on devices other than" }

 !$omp groupprivate(x, z) device_Type( any )
 !$omp declare target local(x) device_type ( any )
 !$omp declare target enter( ii) ,local(y), device_type ( host )
 !$omp groupprivate(y) device_type( host)
 !$omp groupprivate(v) device_type (nohost )
 !$omp groupprivate(u)

 !$omp groupprivate(k)
end module

File: gcc/testsuite/c-c++-common/gomp/groupprivate-1.c
/* { dg-do compile }  */

int x, y, z, a, b;
#pragma omp groupprivate(a,b)  // { dg-message "sorry, unimplemented: 'omp groupprivate'" }
#pragma omp groupprivate(x) device_type(any)     // { dg-message "sorry, unimplemented: 'omp groupprivate'" }
#pragma omp groupprivate(y) device_type(host)    // { dg-message "sorry, unimplemented: 'omp groupprivate'" }
#pragma omp groupprivate(z) device_type(nohost)  // { dg-message "sorry, unimplemented: 'omp groupprivate'" }

[[omp::decl (groupprivate)]] int d, e;  // { dg-message "sorry, unimplemented: 'omp groupprivate'" }
[[omp::decl (groupprivate,device_type(any))]] int f1;     // { dg-message "sorry, unimplemented: 'omp groupprivate'" }
[[omp::decl (groupprivate,device_type(host))]] int f2;    // { dg-message "sorry, unimplemented: 'omp groupprivate'" }
[[omp::decl (groupprivate,device_type(nohost))]] int f3;  // { dg-message "sorry, unimplemented: 'omp groupprivate'" }
